### DISCUSSION TITLE
New id is not generated for new entry

### DISCUSSION THREAD

**Discussion by donacrisante** [neuefische/questions#700] [Web] [answered]
Hello everyone,
for some reason no new id is generated when I enter a new trip in my form. It's honestly a mystery to me why, as I think the function is written correctly:

function handleFormSubmit(newEntry) {    
    setEntries([{ id: uid(), ...newEntry }, ...entries]);
    router.push("/journey-list");
  }

A new id was always generated before and now somehow not anymore. Does anyone have a moment to look over this?

Here my PR:
https://github.com/donacrisante/Capstone-project/pull/11

Many thanks in advance!

**Comment by Kheirah** [comment, marked as answer]
Since you are assigning an id in the submit function, you will need to change the order of your spread.
```js
function handleFormSubmit(newEntry) {
    setEntries([{ ...newEntry, id: uid() }, ...entries]);
    router.push("/journey-list");
}
```

**Reply by donacrisante** [reply]
Thank you Ernst, now it works.
